### ISSUE TITLE
Calling .transform on a parsed string does not give the new Tree the transformed syntax

### ISSUE THREAD

**Issue by hugo-vrijswijk** [scalameta/scalameta#1526]
When a string is parsed with .parse[U], and then transformed, the new transformed tree does not have the .syntax it should have, but instead the syntax of the old tree. 

Code example: 
```scala
val original = "def foo = 4".parse[Stat].get

val transformed = original.transform({case q"4" => q"5"})

println(transformed.syntax) // 'def foo = 4', should be 'def foo = 5'
println(transformed.syntax == original.syntax) // true
println(transformed.structure == original.structure) // false
```
If the original parsed was a Source, then calling .asInstanceOf[Source].copy() seems to fix the syntax.

Scala 2.12.6, Scalameta 3.7.4

**Comment by olafurpg** [comment]
Thanks for reporting! I am able to reproduce, this seems to be a regression from https://github.com/scalameta/scalameta/pull/1349 I am unable to reproduced in v3.3.0, so you can use that version if you prefer (there haven't been any other changes to the tree API 3.3-3.7 that i'm aware of).

The root cause of this bug is https://github.com/scalameta/scalameta/blob/afbd949b70144d86973ac9c573474d4b777fa007/scalameta/trees/shared/src/main/scala/scala/meta/internal/prettyprinters/TreeSyntax.scala#L597 The correct solution is to fix https://github.com/scalameta/scalameta/issues/1350 introducing `Origin.Transformed` instead of `Origin.Parsed`. @MasseGuillaume any chance you could take a look at this?

**Comment by hugo-vrijswijk** [comment]
Thanks for the quick reply! I'll use 3.3.0 for now

**Comment by MasseGuillaume** [comment]
@olafurpg I propose we revert #1349. This will leave us some time to come up with a plan for tree transformation in scala-syntax.

**Comment by olafurpg** [comment]
Fixed in #1527. Will be released as 4.0.0-M4-115-6c5f231b-SNAPSHOT as soon as CI completes and will be part of 4.0.0-M5 before the weekend. Sorry for the pain this caused!

**Comment by hugo-vrijswijk** [comment]
Awesome, thanks!
